File: src/main/java/to7fa/entities/participation_evenement.java
package to7fa.entities;

import java.sql.Date;

/**
 *
 * @author Admin
 */
public class participation_evenement {

    private int ID_participation;
    private int ID_user;
    private int ID_event;
    private int nombre_participation;
    private String num_tel;
    private String username; // Added field for username

    public participation_evenement() {
    }

    public participation_evenement(int ID_participation, int ID_user, int ID_event, int nombre_participation, String num_tel) {
        this.ID_participation = ID_participation;
        this.ID_user = ID_user;
        this.ID_event = ID_event;
        this.nombre_participation = nombre_participation;
        this.num_tel = num_tel;
        this.username = username; // Initialize username in constructor
    }

    public participation_evenement(int ID_user, int ID_event, int nombre_participation, String num_tel) {
        this.ID_user = ID_user;
        this.ID_event = ID_event;
        this.nombre_participation = nombre_participation;
        this.num_tel = num_tel;
    }
    // Getter and Setter for username
    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }
    public int getID_participation() {
        return ID_participation;
    }

    public void setID_participation(int ID_participation) {
        this.ID_participation = ID_participation;
    }

    public int getID_user() {
        return ID_user;
    }

    public void setID_user(int ID_user) {
        this.ID_user = ID_user;
    }

    public int getID_event() {
        return ID_event;
    }

    public void setID_event(int ID_event) {
        this.ID_event = ID_event;
    }


    public int getNombre_participation() {
        return nombre_participation;
    }

    public void setNombre_participation(int nombre_participation) {
        this.nombre_participation = nombre_participation;
    }

    public String getNum_tel() {
        return num_tel;
    }

    public void setNum_tel(String num_tel) {
        this.num_tel = num_tel;
    }

    @Override
    public String toString() {
        return "participation_evenement{" + "ID_participation=" + ID_participation + ", ID_user=" + ID_user + ", ID_event=" + ID_event + ", nombre_participation=" + nombre_participation + ", num_tel=" + num_tel +  ", username='" + username + '\'' +'}';
    }








}

File: src/Entity/ParticipationEvenement.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * ParticipationEvenement
 *
 * @ORM\Table(name="participation_evenement", indexes={@ORM\Index(name="fk_event_participation", columns={"id_event"})})
 * @ORM\Entity
 */
class ParticipationEvenement
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_participation", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idParticipation;

    /**
     * @var int|null
     *
     * @ORM\Column(name="nombre_participation", type="integer", nullable=true)
     */
    private $nombreParticipation;

    /**
     * @var string|null
     *
     * @ORM\Column(name="num_tel", type="string", length=255, nullable=true)
     */
    private $numTel;

    

    /**
     * @var \Evenement
     *
     * @ORM\ManyToOne(targetEntity="Evenement")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_event", referencedColumnName="id_event")
     * })
     */
    private $idEvent;

    public function getIdParticipation(): ?int
    {
        return $this->idParticipation;
    }

    public function getNombreParticipation(): ?int
    {
        return $this->nombreParticipation;
    }

    public function setNombreParticipation(?int $nombreParticipation): self
    {
        $this->nombreParticipation = $nombreParticipation;

        return $this;
    }

    public function getNumTel(): ?string
    {
        return $this->numTel;
    }

    public function setNumTel(?string $numTel): self
    {
        $this->numTel = $numTel;

        return $this;
    }

    /* public function getIdUser(): ?User
    {
        return $this->idUser;
    }

    public function setIdUser(?User $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    } */

    public function getIdEvent(): ?Evenement
    {
        return $this->idEvent;
    }

    public function setIdEvent(?Evenement $idEvent): self
    {
        $this->idEvent = $idEvent;

        return $this;
    }


}
